scheduler/src/cook/kubernetes/api.clj

```diff
@@ -0,0 +1,316 @@
+                                                             container-requests (map (fn [^V1Container c]
+                                                                                       (-> c
+                                                                                           .getResources
+                                                                                           .getRequests
+                                                                                           convert-resource-map))
+                                                                                     containers)]
+                                                         (apply merge-with + container-requests))))
+                                                (apply merge-with +)))
+                                         node-name->pods)]
+    node-name->requests))
+
+(def cook-container-name-for-job
+  "required-cook-job-container")
+
+(defn task-metadata->pod
+  [{:keys [task-id command container task-request hostname]}]
+  (let [{:keys [resources]} task-request
+        {:keys [mem cpus]} resources
+        {:keys [docker]} container
+        {:keys [image]} docker
+        pod (V1Pod.)
+        pod-spec (V1PodSpec.)
+        metadata (V1ObjectMeta.)
+        container (V1Container.)
+        env (map (fn [[k v]]
+                   (let [env (V1EnvVar.)]
+                     (.setName env (str k))
+                     (.setValue env (str v))
+                     env))
+                 (:environment command))
+        resources (V1ResourceRequirements.)]
+    ; metadata
+    (.setName metadata (str task-id))
+
+    ; container
+    (.setName container cook-container-name-for-job)
+    (.setCommand container
+                 ["/bin/sh" "-c" (:value command)])
+
+    (.setEnv container (into [] env))
+    (.setImage container image)
+
+    (.putRequestsItem resources "memory"
+                      (Quantity. (BigDecimal/valueOf ^double (* 1024 1024 mem))
+                                 Quantity$Format/DECIMAL_SI))
+    (.putLimitsItem resources "memory"
+                    (Quantity. (BigDecimal/valueOf ^double (* 1024 1024 mem))
+                               Quantity$Format/DECIMAL_SI))
+    (.putRequestsItem resources "cpu"
+                      (Quantity. (BigDecimal/valueOf ^double cpus)
+                                 Quantity$Format/DECIMAL_SI))
+    (.setResources container resources)
+
+    ; pod-spec
+    (.addContainersItem pod-spec container)
+    (.setNodeName pod-spec hostname)
+    (.setRestartPolicy pod-spec "Never")
+
+    ; pod
+    (.setMetadata pod metadata)
+    (.setSpec pod pod-spec)
+
+    pod))
+;;
+;;  This file contains accessor functions for all kubernetes API. That way we can conveniently mock these functions in clojure.
+;;
+
+(defn TODO
+  []
+  (throw (UnsupportedOperationException. "TODO")))
```

Comment:
Is this used anywhere? Can we delete it?

Reply:
Gone.